### ISSUE TITLE
Callable/Configurable poemgame

### ISSUE THREAD

**Issue by ThePotatoGuy** [Backdash/MonikaModDev#427]
WIP. Since the poem game is a built in minigame, why not use it?

Currently working on generalizing this.

EDIT: Heres a sneak peek at the amount of configurability: (Will update overtime)
```
# ACT 1 ------------------
# silghty configrable poemgame call using mostly ACT 1 params.
# 
# IN:
# NOTE: SEE mas_poem_minigame documentation for full descriptions of the params
#   gather_words - (Default: False)
#   music_fadein - (Default: 2.0)
#   music_fadeout - (Default: 2.0)
#   music_filename - (Default: audio.t4 - the default poem game audio)
#   only_winner - (Default: False)
#   poem_wordlist - (Default: None)
#   sel_sound - (Default: gui.activate_sound)
#   show_monika - (Default: False)
#   show_natsuki - (Default: True)
#   show_poemhelp - (Default: True)
#   show_sayori - (Default: True)
#   show_yuri - (Default: True)
#   total_words - (Default: 20)
#   trans_fast - (Default: False)
#   trans_in - (Default: True)
#   trans_out - (Default: True)
#
# OUT:
#   See mas_poem_minigame -> STOCK_MODE documentation
#

# ACT 2 ------------------
# slightly configurable poemgame call using mostly ACT 2 params.
#
# IN:
# NOTE: SEE mas_poem_minigame documentation for full descriptions of the params
#   gather_words - (Default: False)
#   glitch_baa - (Default: (True, None, False))
#   glitch_words - (Default: None)
#   glitch_wordscare - (Default: (True, None))
#   glitch_wordscare_sound - (Default: (True, None))
#   music_fadein - (Default: 0.0)
#   music_fadeout - (Default: 0.0)
#   music_filename - (Default: audio.t4)
#   one_counter - (Default: True)
#   only_winner - (Default: False)
#   poem_wordlist - (Defualt: None)
#   sel_sound - (Default: gui.activate_sound)
#   show_eyes - (Default: (True, None))
#   show_monika - (Defualt: False)
#   show_natsuki - (Default: True)
#   show_poemhelp - (Default: True)
#   show_sayori - (Default: True)
#   show_yuri - (Default: True)
#   show_yuri_cut - (Default: True)
#   show_yuri_scary - (Default: (True, None, False))
#   total_words - (Default: 20)
#   trans_fast - (Default: False)
#   trans_in - (Default: False)
#   trans_out - (Default: True)
#
# OUT:
#   see mas_poem_minigame -> STOCK_MODE documentation

# ACT 3 ------------------ (MONIKA MODE~~~~~~~~~~~~~~)
# slightly configurable poemgame call using mostly ACT 3 params
#
# IN:
# NOTE: SEE mas_poem_minigame_documentation for full descriptions of the params
#   gather_words - (Default: False)
#   glitch_nb - (Default: True)
#   glitch_words - (Default: (True, None, None))
#   hop_monika - (Default: False)
#   music_fadein - (Default: 0.0)
#   music_fadeout - (Default: 0.0)
#   music_filename - (Default: audio.ghostmenu)
#   total_words - (Default: 20)
#   trans_fast - (Default: False)
#   trans_in - (Default: False)
#   trans_out - (Default: True)
#
# OUT:
#   SEE mas_poem_minigame -> MONIKA_MODE documentation

# THE BIG ONE ------------
# completely configurable poemgame call.
# NOTE: well, almost completely configurable
# NOTE: The CALLER is responsible:
#   WHEN ENTERING THIS LABEL:
#       - fading out/stopping music
#   WHEN LEAVING THIS LABEL:
#       - fading in/starting music
#       - fading in/starting visuals/scenes
#
# IN:
#   (REQUIRED)
#   flow - what are we trying to do with this minigame. This changes the
#       returned values: 
#
#       (these are from mas_poemgame_consts)
#       DISPLAY_MODE - glitch/display mode.
#           If gather_words is True:
#               Returns list of words selected
#           else:
#               Nothing
#
#       STOCK_MODE - stock minigame mode. 
#           If only_winner is True:
#               Returns only the winner and points as Tuple:
#                   [0] -> name of winner (defined in mas_poemgame_consts)
#                   [1] -> pts they won
#               If gather_words is True:
#                   [2] -> list of words selected
#           else:
#               Returns the point totals for each character in a dict of the 
#               following format: (keys are defined in mas_poemgame_consts)
#                   "sayori": <pts>
#                   "natsuki": <pts>
#                   "yuri": <pts>
#                   "monika": <pts>
#               If gather _words is True:
#                   "words": list of words selected
#
#       MONIKA_MODE - poemgame as it was in Act 3. Slightly configurable
#           If gather_words is True:
#               Returns list of words selected
#           else:
#               Nothing
#
#       TODO: add more flows? when we need them ofc
#
#       NOTE: If the given flow is none of the above, DISPLAY_MODE is assumed
#
#   (CONFIGURATION OPTIONS): (ORDER DOES NOT MATCH INPUT PARAMS)
#
#   gather_words - True will include a list of the selected words in the 
#       return, False will not
#       NOTE: The types of objects in this list may be different:
#           If we are in monika_mode, then this will be a list of 
#               MASPoemWord objects
#           If poem_wordlist is not given, then this will be a list of
#               PoemWord objects
#           If poem_wordlist is given, then this will be a list of 
#               MASPoemWord Objects
#       NOTE: The returned objects are deepcopied
#       (Default: False)
#
#   glitch_baa - Tuple of the following format:
#           [0] -> True means do the baa glitch, false means no
#               (If None, default False)
#           [1] -> odds that the baa glitch occurs (1 out of x)
#               (If None, default 10)
#           [2] -> True means play the baa glitch more than once, False means
#               only once
#               (If None, default False)
#       If the tuple is None (or is not of length 2), the default values are
#       used.
#       NOTE: Only is active if glitch_wordscare is used.
#       (Default: None)
#
#   glitch_nb - True will use the glitched notebook. False will use regular
#       one.
#       (Default: False)   
#
#   glitch_words - Tuple of the following format:
#           [0] -> True means display gltich words, false means no
#               (If None, default False)
#           [1] -> odds that a space appears instead of a letter (1 out of x)
#               (If None, default is in glitchWord())
#           [2] -> odds that a nonunicode appears instead of a letter (1 out 
#               of x)
#               (If None, default is in glitchWord())
#       If the tuple is None (or is not of length 3), the default values are
#       used. (SEE glitchWord())
#       NOTE: This effect is purely visible. If you want the glitchable words
#       that cause the scare, use glitch_wordscare
#       (Default: None)
#
#   glitch_wordscare - Tuple of the following format:
#           [0] -> True means display glitch words that cause the scare
#               (If None, default False)
#           [1] -> odds that one of these glitch words will appear (1 out of x)
#               (If None, default is 400)
#       If the tuple is None (or is not of length 2), the default values are
#       used.
#       NOTE: This is the glitch that causes a scare. If you want purely
#       visisble glitch words, use glitch_words
#       (Default: None)
#
#   glitch_wordscare_sound - Tuple of the following format:
#           [0] -> True means play the glitch sound if a glictch word was
#               clicked, False will not
#               (If None, default True)
#           [1] -> odds that a glitch sound will play (1 out of x)
#               NOTE: these odds are applied after glitch_baa, if that is True
#               (If None, default 2)
#       If the tuple is None (or is not of length 2), the deafult vlaues are
#       used.
#       NOTE: Only active if glitch_wordscare is used.
#       (Default: None)
#
#   hop_monika - True means monika will hop when a word is selected in monika
#       mode. False will not
#       NOTE: Only applies to Monika mode hopping
#       (Default: False)
#
#   music_fadein - Number of seconds to fade in music at start of poem(float)
#       (Default: 2.0)
#
#   music_fadeout - Number of seconds to fade out music at end of poem(float) 
#       (Default: 2.0)
#
#   music_filename - filename of the music to play. 
#       NOTE: Set to the string "BACK" to use background/currently playing 
#           music. 
#       NOTE: Setting this to None will stop playing music entirely
#       NOTE: NO checks are made for the existence of this file. Please no
#       bully.
#       (Default: audio.t4 - the defualt poem game audio)
#
#   one_counter - True will apply the glitch that made the poem counter 
#       display 1 over and over again instead of the approriate number. False
#       will do regular counting
#       (Default: False)
#
#   only_winner - True will the return value of STOCK_MODE the following tuple:
#           [0] -> name of winning girl (defined in mas_poemgame_consts)
#           [1] -> the amount of points this girl received
#       False will return what is described above in flow.
#       (Default: False)
#
#   poem_wordlist - MASPoemWordList of the poemwords to use this game.
#       If None, the default full_wordlist defined in script-poemgame is used.
#       NOTE: If this is None, monika will be visible but she will NOT receive
#           points nor hop. This is to ensure compatibility with the stock 
#           PoemWord class
#
#   sel_sound - Filename of the sound to play when a word is selected
#       (Default: gui.activate_sound)
#
#   show_eyes - Tuple of the following format:
#           [0] -> True displays the eyes glitch, False does not
#               (Default: False)
#           [1] -> odds that the eyes glitch appears (1 out of x)
#               (Default: 5)
#       If Tuple is None (or not of length 2), the default values are used
#       (Default: None)
#
#   show_monika - True will display monika and her related actions. False will
#       not. 
#       NOTE: If poem_wordlist is None, monika will be visible but she will
#           NOT hop nor will she get points
#       (Default: True)
#
#   show_natsuki - True will display natsuki and her related actions. False 
#       will not.
#       (Default: False)
#
#   show_poemhelp - True will display the poem help screen. False will not
#       (Default: False)
#
#   show_sayori - True will display sayori and her related actions. False will
#       not.
#       (Default: False)
#
#   show_yuri - True will display yuri and her related actions. False will not.
#       (Default: False)
#
#   show_yuri_cut - True will display a cut yuri instead of regular yuri,
#       False will not.
#       NOTE: show_yuri must be true for this to work
#       (Defualt: False)
#
#   show_yuri_scary - Tuple of the following format:
#           [0] -> True will show the scary yuri sticker, False will not
#               (If none, Default False)
#           [1] -> odds that the scary yuri sticker should show up (1 out of x)
#               (If None, Default 100)
#           [2] -> True means show the sticker more than once, False means
#               only once
#               (If None, Default Flase)
#       If the tuple is None (or not of length 3), the default values are used
#       NOTE: Only active if show_yuri is True
#       (Default: None)
#
#   total_words - Number of words that can be picked this game
#       (Default: 20)
#
#   trans_fast - True will use a faster transition sequence. False will use
#       stock transition.
#       NOTE: This transition is about twice as fast as the stock one
#       NOTE: This only applies if trans_in or trans_out is TRUE
#       (Default: False)
#
#   trans_in - True will perform a fade-in transition at start. False will not
#       (Default: True)
#
#   trans_out - True will perform a fade-out transition at start.
#       False will not.
#       (Default: True)
#
#   (DISPLAY_MODE OPTIONS)
#   gather_words
#   glitch_baa
#   glitch_nb
#   glitch_words
#   glitch_wordscare
#   glitch_wordscare_sound
#   music_fadein
#   music_fadeout
#   music_filename
#   one_counter
#   poem_wordlist
#   sel_sound
#   show_eyes
#   show_monika
#   show_natsuki
#   show_poemhelp
#   show_sayori
#   show_yuri
#   show_yuri_cut
#   show_yuri_scary
#   total_words
#   trans_fast
#   trans_in
#   trans_out
#
#   (STOCK_MODE OPTIONS)
#   gather_words
#   glitch_baa
#   glitch_nb
#   glitch_words
#   glitch_wordscare
#   glitch_wordscare_sound
#   music_fadein
#   music_fadeout
#   music_filename
#   one_counter
#   only_winner
#   poem_wordlist
#   sel_sound
#   show_eyes
#   show_monika
#   show_natsuki
#   show_poemhelp
#   show_sayori
#   show_yuri
#   show_yuri_cut
#   show_yuri_scary
#   total_words
#   trans_fast
#   trans_in
#   trans_out
#
#   (MONIKA_MODE OPTIONS)
#   gather_words
#   glitch_baa
#   glitch_nb
#   glitch_words
#   glitch_wordscare
#   glitch_wordscare_sound
#   hop_monika
#   music_fadein
#   music_fadeout
#   music_filename
#   one_counter
#   sel_sound
#   show_eyes
#   show_poemhelp
#   total_words
#   trans_fast
#   trans_in
#   trans_out
#
# OUT:
#   DISPLAY_MODE:
#       If gather_words is True:
#           Returns of list of words selected
#       else:
#           None
#
#   STOCK_MODE:
#       If only_winner is True:
#           Returns only the winner and points as Tuple:
#               [0] -> name of winner (defined in mas_poemgame_consts)
#               [1] -> pts they won
#           If gather_words is True:
#               [2] -> list of words selected
#       else:
#           Returns the point toals for each character in a dict of the
#           following format: (keys are defined in mas_poemgame_consts)
#               "sayori": <pts>
#               "natsuki": <pts>
#               "yuri": <pts>
#               "monika": <pts>
#           If gather_words is True:
#               "words": list of words selected
#
#   MONIKA_MODE:
#       If gather_words is True:
#           Returns list of words selected
#       else:
#           None
#
```

**Comment by Kiosho** [comment]
Nice! Maybe we could have Monika write more poems as well. Future topics though. Bring back the poems!

**Comment by ThePotatoGuy** [comment]
This probably wont be available until after 0.7.0. We'll see.

**Comment by miranimal** [comment]
Is there any way possible to be able to have the option to play the mod without jump scares or glitches? I don't think I can handle anymore of those.
